cmd/addon-operator-manager/main.go

```diff
@@ -42,29 +45,65 @@ func init() {
 }
 
 type options struct {
-	metricsAddr           string
-	pprofAddr             string
-	enableLeaderElection  bool
-	enableMetricsRecorder bool
-	probeAddr             string
+	metricsAddr                 string
+	pprofAddr                   string
+	enableLeaderElection        bool
+	enableMetricsRecorder       bool
+	enableMetricsTLSTermination bool
+	probeAddr                   string
+	metricsTlsCertDir           string
+	nonTLSMetricsAddr           string
+	metricsTlsConfig            tlsConfig
+}
+
+type tlsConfig struct {
+	caCertPath string
+	certPath   string
+	keyPath    string
 }
 
 func parseFlags() *options {
 	opts := &options{}
 
-	flag.StringVar(&opts.metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&opts.metricsAddr, "metrics-addr", ":8080", "The address at which metrics (https or http) will be exposed.")
 	flag.StringVar(&opts.pprofAddr, "pprof-addr", "", "The address the pprof web endpoint binds to.")
 	flag.BoolVar(&opts.enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
 	flag.StringVar(&opts.probeAddr, "health-probe-bind-address", ":8081",
 		"The address the probe endpoint binds to.")
 	flag.BoolVar(&opts.enableMetricsRecorder, "enable-metrics-recorder", true, "Enable recording Addon Metrics")
+	flag.BoolVar(&opts.enableMetricsTLSTermination, "enable-metrics-tls-termination", true, "Enable metrics endpoint to be TLS-terminated")
+	flag.StringVar(&opts.metricsTlsCertDir, "metrics-tls-cert-dir", "/tmp/k8s-metrics-server/serving-certs/", "Path to the directory where the TLS config-related files exist (ca.crt, tls.crt, tls.key)")
 	flag.Parse()
 
 	return opts
 }
 
+func preprocessOpts(opts *options) {
+	// opts.nonTLSMetricsAddr - where the actual controller-runtime metric server will be setup
+	// opts.metricsAddr - where the relay server will be setup if --enable-metrics-tls-termination flag would be provided
+	// For the end-user/client, if --enable-metrics-tls-termination is provided, https://<host>:<metrics-addr>/metrics will be reachable
+	// else, http://<host>:<metrics-addr>/metrics will be reachable
+	if opts.enableMetricsTLSTermination {
+		opts.nonTLSMetricsAddr = defaultNonTlsMetricsAddr
+	} else {
+		opts.nonTLSMetricsAddr = opts.metricsAddr
+	}
```

Comment:
I feel like things can be simplified a bit. We can accept:
- `--metrics-addr` -> this will be for /metrics exposed via HTTP
- `--metrics-tls-relay-addr` -> this will be the address of the relay server

If relay server port is provided, then setup the relay server, else skip it (in the previous case, we set a default on this address if it wasn't provided).

This way, we can configure both ports via command-line and have one less boolean flag. Let me know what you think :)

Reply:
This was exactly the previous approach but the current approach is even simpler for the client/end-user. They don't have to configure/deal with/know about multiple ports of metrics exposure (http/https). All they end up getting is a single metrics endpoint everytime and they only need to care is calling it via http or https, depending on the --enable-metrics-tls-termination flag (enabled by default). Plus, this semantically makes sense with `opts.metricsAddr`'s purpose which is solely meant to be the address/port which exposes the metrics from the eyes of the client.

From implementation's perspective, the LOC remains the same too and the number of flags/args remain the same too i.e. 2 (--metrics-addr and --enable-metrics-tls-termination/--metrics-tls-relay-addr)